rdr_service/dao/genomics_dao.py

```diff
@@ -1217,6 +1218,83 @@ def get_members_from_biobank_ids(self, biobank_ids, genome_type=None):
                 )
             return members.all()
 
+    def get_array_members_files_available(self, sample_list=None):
+        with self.session() as session:
+            subquery = session.query(
+                GenomicSetMember.id,
+                func.count(GenomicGcDataFile.file_type).label("file_count")
+            ).join(
+                GenomicGCValidationMetrics,
+                GenomicGCValidationMetrics.genomicSetMemberId == GenomicSetMember.id
+            ).outerjoin(
+                GenomicGcDataFile,
+                GenomicGcDataFile.identifier_value == GenomicGCValidationMetrics.chipwellbarcode
+            ).filter(
+                GenomicSetMember.genomicWorkflowState == GenomicWorkflowState.GC_DATA_FILES_MISSING,
+                GenomicSetMember.genomeType == 'aou_array',
+                GenomicSetMember.ignoreFlag != 1,
+                GenomicGCValidationMetrics.ignoreFlag != 1,
+                GenomicGcDataFile.ignore_flag != 1,
+                GenomicGcDataFile.file_type.in_(
+                    [file_type['file_type'] for file_type in array_file_types_attributes
+                     if file_type['required']])
```

Comment:
It may help readability to set a variable higher up for `required_array_file_types`. You can then use it here and in calculating the len(). I have a hard time mentally computing list comprehensions :)

Reply:
I've added a variable for the list comprehension.